### PR TITLE
Base Mainnet deployment

### PR DESCRIPTION
**Pull request by julianmrodri**
Base deployment scripts and addresses (protocol and Plugins)

Also includes deployment of aETHUSDC (Aave V3 USDC) to Mainnet Ethereum.

### CODE REVIEW

File: scripts/deployment/phase2-assets/collaterals/deploy_ctokenv3_usdbc_collateral.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const collateral = <CTokenV3Collateral>await CTokenV3Factory.connect(deployer).deploy(
+    {
+      priceTimeout: priceTimeout.toString(),
+      chainlinkFeed: networkConfig[chainId].chainlinkFeeds.USDC,
+      oracleError: usdcOracleError.toString(),
+      erc20: erc20.address,
+      maxTradeVolume: fp('1e6').toString(), // $1m,
+      oracleTimeout: oracleTimeout(chainId, usdcOracleTimeout).toString(), // 24h hr,
+      targetName: hre.ethers.utils.formatBytes32String('USD'),
+      defaultThreshold: fp('0.0125').toString(), // 1% + 0.25%
```
Comment (tbrent) [inline scripts/deployment/phase2-assets/collaterals/deploy_ctokenv3_usdbc_collateral.ts:76]: 0.013

**Review by tbrent — COMMENTED** [review]
found a few `oracleErrors`/`oracleTimeouts` that were off

I also neglected to mention before (my b) that the `defaultThreshold` should be a function of the oracleError. For fiatcoins it's 0.01 + `oracleError` and for LSDs its 0.02 + `oracleError`

File: scripts/deployment/phase2-assets/collaterals/deploy_cbeth_collateral.ts [hunk 2]
```diff
@@ -69,23 +74,24 @@ async function main() {
...
     collateral = await CBETHCollateralFactory.connect(deployer).deploy(
       {
         priceTimeout: priceTimeout.toString(),
         chainlinkFeed: networkConfig[chainId].chainlinkFeeds.ETH!,
-        oracleError: oracleError.toString(), // 0.5% & 2%,
+        oracleError: oracleError.toString(), // 0.15% & 0.5%,
         erc20: networkConfig[chainId].tokens.cbETH!,
         maxTradeVolume: fp('1e6').toString(), // $1m,
-        oracleTimeout: oracleTimeout(chainId, '3600').toString(), // 1 hr,
+        oracleTimeout: oracleTimeout(chainId, '1200').toString(), // 20 min
```
[the remark this answers is not in the document]
Comment (julianmrodri) [reply scripts/deployment/phase2-assets/collaterals/deploy_cbeth_collateral.ts:92]: This one is for the ETH/USD feed
image

**Comment by julianmrodri** [comment]
> found a few `oracleErrors`/`oracleTimeouts` that were off
> 
> I also neglected to mention before (my b) that the `defaultThreshold` should be a function of the oracleError. For fiatcoins it's 0.01 + `oracleError` and for LSDs its 0.02 + `oracleError`

Implemented all the fixes. I had to redeploy most collaterals due to the `defaultThreshold` change. Already updated the addresses in the PR.

**Comment by tbrent** [comment]
Awesome, LGTM

File: scripts/deployment/phase2-assets/collaterals/deploy_aave_v3_usdc.ts [hunk 3]
```diff
@@ -49,7 +45,10 @@ async function main() {
   /********  Deploy Aave V3 USDC wrapper  **************************/
 
   const StaticATokenFactory = await hre.ethers.getContractFactory('StaticATokenV3LM')
```
Comment (pmckelvy1) [inline scripts/deployment/phase2-assets/collaterals/deploy_aave_v3_usdc.ts:47]: lets add a check in this file that we're targetting mainnet (just like we have a check in the ausdbc file that we're targetting base

File: scripts/deployment/phase2-assets/collaterals/deploy_ctokenv3_usdc_collateral.ts [hunk 4]
```diff
@@ -40,13 +40,12 @@ async function main() {
 
   const deployedCollateral: string[] = []
 
-  /********  Deploy CompoundV3 USDC - cUSDCv3  **************************/
+  /********  Deploy CompoundV3 USDC - cUSDCv3 **************************/
 
   const WrapperFactory: ContractFactory = await hre.ethers.getContractFactory('CusdcV3Wrapper')
-
```
Comment (pmckelvy1) [inline scripts/deployment/phase2-assets/collaterals/deploy_ctokenv3_usdc_collateral.ts:46]: lets check for mainnet in this file (we have a separate usdbc file for base)

File: scripts/deployment/phase2-assets/collaterals/deploy_aave_v3_usdc.ts (hunk 3 above)
Comment (julianmrodri) [reply scripts/deployment/phase2-assets/collaterals/deploy_aave_v3_usdc.ts:52]: But cant USDC be deployed at some point to Base?

File: scripts/deployment/phase2-assets/collaterals/deploy_ctokenv3_usdc_collateral.ts (hunk 4 above)
Comment (julianmrodri) [reply scripts/deployment/phase2-assets/collaterals/deploy_ctokenv3_usdc_collateral.ts:46]: Also its not just Mainnet, should be anything besides L2 Base (like tenderly should work as well, hh fork, etc)

**Comment by julianmrodri** [comment]
Can we merge this one?

**Pull request merged**